chef/cookbooks/provisioner/recipes/setup_base_images.rb

```diff
@@ -420,197 +420,206 @@
+
+    if require_install_dir
+      # Don't bother for OSes that are not actually present on the provisioner node.
+      next unless File.directory?(os_dir) && File.directory?("#{os_dir}/install")
+    end
+
+    # Index known barclamp repositories for this OS
+    node[:provisioner][:repositories][os] ||= Mash.new
+    node[:provisioner][:repositories][os][arch] ||= Mash.new
+
+    if File.exist?("#{os_dir}/crowbar-extra") && File.directory?("#{os_dir}/crowbar-extra")
+      Dir.foreach("#{os_dir}/crowbar-extra") do |f|
+        next unless File.symlink? "#{os_dir}/crowbar-extra/#{f}"
+        node[:provisioner][:repositories][os][arch][f] ||= Hash.new
+        case
+        when os =~ /(ubuntu|debian)/
+          bin = "deb #{web_path}/crowbar-extra/#{f} /"
+          src = "deb-src #{web_path}/crowbar-extra/#{f} /"
+          node.set[:provisioner][:repositories][os][arch][f][bin] = true if
+            File.exist? "#{os_dir}/crowbar-extra/#{f}/Packages.gz"
+          node.set[:provisioner][:repositories][os][arch][f][src] = true if
+            File.exist? "#{os_dir}/crowbar-extra/#{f}/Sources.gz"
+        when os =~ /(redhat|centos|suse)/
+          bin = "baseurl=#{web_path}/crowbar-extra/#{f}"
+          node.set[:provisioner][:repositories][os][arch][f][bin] = true
         else
           raise ::RangeError.new("Cannot handle repos for #{os}")
         end
+      end
     end
-  end
 
-  # If we were asked to use a serial console, arrange for it.
-  if node[:provisioner][:use_serial_console]
-    append << " console=tty0 console=#{node[:provisioner][:serial_tty]}"
-  end
+    # If we were asked to use a serial console, arrange for it.
+    if node[:provisioner][:use_serial_console]
+      append << " console=tty0 console=#{node[:provisioner][:serial_tty]}"
+    end
 
-  # Make sure we get a crowbar install key as well.
-  unless crowbar_key.empty?
-    append << " crowbar.install.key=#{crowbar_key}"
-  end
+    # Make sure we get a crowbar install key as well.
+    unless crowbar_key.empty?
+      append << " crowbar.install.key=#{crowbar_key}"
+    end
 
-  # These should really be made libraries or something.
-  case
-  when /^(suse)/ =~ os
-    # Add base OS install repo for suse
-    node.set[:provisioner][:repositories][os]["base"] = { "baseurl=http://#{admin_ip}:#{web_port}/#{os}/install" => true }
+    # These should really be made libraries or something.
+    case
+    when /^(suse)/ =~ os
+      # Add base OS install repo for suse
+      node.set[:provisioner][:repositories][os][arch]["base"] = { "baseurl=#{admin_web}" => true }
 
-    ntp_servers = search(:node, "roles:ntp-server")
-    ntp_servers_ips = ntp_servers.map { |n| Chef::Recipe::Barclamp::Inventory.get_network_by_type(n, "admin").address }
+      ntp_servers = search(:node, "roles:ntp-server")
+      ntp_servers_ips = ntp_servers.map { |n| Chef::Recipe::Barclamp::Inventory.get_network_by_type(n, "admin").address }
```

Comment:
Metrics/LineLength: Line is too long. [121/100] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)